src/components/forms/application/v2/bank-account-connection.tsx

```diff
@@ -27,6 +36,404 @@ const TRUSTED_FLINKS_ORIGIN = new URL(
+        </p>
+      </div>
+
+      {/* What happens next */}
+      <div className="space-y-4">
+        <h3 className="text-xl font-semibold text-gray-900">
+          What happens next:
+        </h3>
+
+        <div className="space-y-3">
+          {nextSteps.map((step, index) => (
+            <div
+              key={index}
+              className="bg-green-50 border border-green-100 rounded-2xl p-4 text-left"
+            >
+              <div className="flex items-start gap-3">
+                <div className="flex-shrink-0 w-8 h-8 bg-green-100 rounded-full flex items-center justify-center">
+                  <span className="text-green-700 font-bold text-sm">
+                    {index + 1}
+                  </span>
+                </div>
+                <p className="text-gray-700 text-sm font-medium pt-1">{step}</p>
+              </div>
+            </div>
+          ))}
+        </div>
+      </div>
+    </div>
+  );
+};
+
+const FlinksConnectionScreen: React.FC<{
+  flinksConnectionParams: URLSearchParams;
+  isLoading: boolean;
+  onBack: () => void;
+}> = ({ flinksConnectionParams, isLoading, onBack }) => {
+  return (
+    <div className="max-w-4xl mx-auto">
+      <div
+        className="bg-white border border-gray-200 rounded-2xl overflow-hidden"
+        style={{ minHeight: '600px' }}
+      >
+        {isLoading && (
+          <div
+            className="flex justify-center items-center flex-col gap-4"
+            style={{ minHeight: '600px' }}
+          >
+            <div className="animate-spin rounded-full h-12 w-12 border-t-2 border-b-2 border-green-600"></div>
+            <p className="text-sm text-gray-500">Connecting to your bank...</p>
+          </div>
+        )}
+        <iframe
+          src={`${process.env.NEXT_PUBLIC_FLINKS_IFRAME_URL}/v2?${flinksConnectionParams.toString()}`}
+          width="100%"
+          height="600"
+          frameBorder="0"
+          style={{ border: 'none', borderRadius: '16px' }}
+          title="Bank Connection"
+        />
+      </div>
+
+      {/* Security note */}
+      <div className="mt-6 bg-green-50 border border-green-100 rounded-2xl p-4">
+        <div className="flex items-start gap-3">
+          <Shield className="h-5 w-5 text-green-600 mt-0.5 flex-shrink-0" />
+          <div>
+            <p className="text-sm font-medium text-green-900 mb-1">
+              Bank-grade security
+            </p>
+            <p className="text-sm text-green-700">
+              Your banking information is encrypted and secure. We cannot see
+              your login credentials.
+            </p>
+          </div>
+        </div>
+      </div>
+
+      <div className="text-center mt-4">
+        <button
+          onClick={onBack}
+          className="text-gray-600 hover:text-gray-800 text-sm underline transition-colors duration-200"
+        >
+          Upload bank statements instead
+        </button>
```

Comment:
Shouldn't this take the user to the manual upload?; it takes them back instead
```suggestion
        <button
          onClick={()=>handleMethodSelect('flinks')}
          className="text-gray-600 hover:text-gray-800 text-sm underline transition-colors duration-200"
        >
          Upload bank statements instead
        </button>
```